src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -182,28 +235,28 @@ public void onDestroyed()
      */
     public void onWakeUp()
     {
-    }
 
+    }
 
     /**
      * Returns whether or not the building has a worker.
      *
      * @return true if building has worker, otherwise false.
      */
-    public boolean hasWorker()
+    public boolean hasEnoughWorkers()
     {
-        return worker != null;
+        return !workers.isEmpty();
     }
 
     /**
-     * Returns if the {@link CitizenData} is the same as {@link #worker}.
+     * Returns if the {@link CitizenData} is the same as the worker.
      *
      * @param citizen {@link CitizenData} you want to compare
      * @return true if same citizen, otherwise false
      */
     public boolean isWorker(final CitizenData citizen)
     {
-        return citizen == worker;
+        return workers.contains(citizen);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "contains()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)